packages/gatsby-plugin-page-creator/src/derive-path.ts

```diff
@@ -1,13 +1,15 @@
 import _ from "lodash"
 import slugify, { Options as ISlugifyOptions } from "@sindresorhus/slugify"
-import { Reporter } from "gatsby"
+import reporter from "gatsby/reporter"
 import {
   extractFieldWithoutUnion,
   extractAllCollectionSegments,
   switchToPeriodDelimiters,
   stripTrailingSlash,
 } from "./path-utils"
 
+type Reporter = typeof reporter
```

Comment:
I'm unsure if there is a better way for this? Maybe also `export { IReporter} from gatsby/reporter`

Reply:
Yes, adding a type export would be good. `gatsby-cli/lib/reporter/reporter` includes a type export for `Reporter`, so you could add `export { Reporter } from "gatsby-cli/lib/reporter/reporter"` inside `reporter.d.ts`